### ISSUE TITLE
Space.GetBody triggers assertion: `path.bodyIndex < m_bodies.size()`

### ISSUE THREAD

**Issue by impaktor** [pioneerspacesim/pioneer#3502]
With this minimal script placed in data/modules, I get a failed assertion/crash on master, from `Space.GetBody()`. 

```lua
local Event = import("Event")
local Space = import("Space")
local Game = import("Game")


local checkStations = function ()

	local dist_ly = 50
	local checksystems = Game.system:GetNearbySystems(dist_ly, function (s) return #s:GetStationPaths() > 0 end)
	assert(#checksystems > 0)

	for _, sys in pairs(checksystems) do
		for __,path in pairs(sys:GetStationPaths()) do
			local body = Space.GetBody(path.bodyIndex)
		end
	end
end

local onGameStart = function ()
    checkStations()
end

Event.Register("onGameStart", onGameStart)
```

Backtrace:
```
CreateCollisionMesh for : (kanara)
WARNING: planetary mass has overflowed! (child of Ceti A)
pioneer: StarSystem.cpp:554: SystemBody* StarSystem::GetBodyByPath(const SystemPath&) const: Assertion `path.bodyIndex < m_bodies.size()' failed.

Program received signal SIGABRT, Aborted.
0x00007ffff52954c9 in raise () from /lib64/libc.so.6
(gdb)
(gdb) bt
#0  0x00007ffff52954c9 in raise () from /lib64/libc.so.6
#1  0x00007ffff5296958 in abort () from /lib64/libc.so.6
#2  0x00007ffff528e296 in __assert_fail_base () from /lib64/libc.so.6
#3  0x00007ffff528e342 in __assert_fail () from /lib64/libc.so.6
#4  0x00000000005deaf1 in StarSystem::GetBodyByPath (this=<optimized out>, path=...)
    at StarSystem.cpp:554
#5  0x00000000005439e9 in Space::FindBodyForPath (this=0x22f1820, path=path@entry=0x7fffffffda40)
    at Space.cpp:393
#6  0x00000000004ad352 in l_space_get_body (l=<optimized out>) at LuaSpace.cpp:574
#7  0x00000000007a3ce5 in luaD_precall (L=L@entry=0xd158f0, func=<optimized out>,
    func@entry=0x22bb1590, nresults=nresults@entry=1) at ldo.c:318
#8  0x00000000007b11a4 in luaV_execute (L=L@entry=0xd158f0) at lvm.c:709
#9  0x00000000007a4309 in luaD_call (L=0xd158f0, func=<optimized out>, nResults=<optimized out>,
    allowyield=0) at ldo.c:395
#10 0x00000000007a34c9 in luaD_rawrunprotected (L=0xd158f0, f=0x79c080 <f_call>, ud=0x7fffffffdd00)
    at ldo.c:131
#11 0x00000000007a46d1 in luaD_pcall (L=L@entry=0xd158f0, func=func@entry=0x79c080 <f_call>,
    u=u@entry=0x7fffffffdd00, old_top=48, ef=<optimized out>) at ldo.c:595
#12 0x00000000007a0ea5 in lua_pcallk (L=0xd158f0, nargs=0, nresults=0, errfunc=<optimized out>,
    ctx=0, k=<optimized out>) at lapi.c:949
#13 0x00000000004b2d9c in pi_lua_protected_call (L=L@entry=0xd158f0, nargs=nargs@entry=0,
    nresults=nresults@entry=0) at LuaUtils.cpp:514
#14 0x0000000000497823 in LuaEvent::Emit () at LuaEvent.cpp:51
#15 0x00000000004d0cea in Pi::StartGame () at Pi.cpp:1045
#16 0x00000000004d678d in Pi::Start () at Pi.cpp:1109
#17 0x0000000000410df9 in main (argc=<optimized out>, argv=<optimized out>) at main.cpp:130
```

**Comment by clausimu** [comment]
Doesn't Space.GetBody() only work for bodies within the current physics space (aka the current system)? Were you expecting a more graceful abort?

Or am I missing something?

I had the same thing happen to me today when working on out-of-system missions for the search & rescue script. But I realized I was trying to get a body for a planet that did not exist yet. I realized the mistake when running the Space.GetBody() function with output from c++, where the m_bodies.size() matches the current system and not the one the planet belongs to.
